Part 2

```diff
@@ -0,0 +1,13 @@
+Matthew Danielson
+6/30/16
+Answers to questions for Part 2 of Assignment 3
+
+1) It is not valid; for one, variable i is not declared; we would have to assume that it has been declared already. The printf statement is also incrrect, as nothing is actually being printed, which will cause an error.
+
+2) This code will not run as intended, since this is an infinite loop, as it will constantly flip between positive and negative plus being constantly reduced by the .5 multiplier.
+
+3) i) n^2
+
+   ii) (coefficient)n^2 
```

Comment:
The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2. 

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4)